compiler/rustc_trait_selection/src/traits/select/mod.rs

```diff
@@ -2626,11 +2627,13 @@ impl<'tcx> ProvisionalEvaluationCache<'tcx> {
     fn on_completion(&self, dfn: usize) {
         debug!(?dfn, "on_completion");
 
+        // If IndexMap gets the `drain_filter` method this could be done in one iteration
```

Comment:
Can we try a perf run with all the other changes except for this?

Reply:
Yes! I changed the branch.